tools/cldr-code/src/main/java/org/unicode/cldr/test/ExampleGenerator.java

```diff
@@ -1094,6 +1096,71 @@ private String handleDayPeriod(XPathParts parts, String value) {
         return formatExampleList(examples.toArray(new String[0]));
     }
 
+    private String handleDateSymbol(XPathParts parts, String value) {
+        // Currently only called for month names, can expand in the future to handle other symbols.
+        // The idea is to show format months in a yMMMM?d date format, and stand-alone months in a
+        // yMMMM? format.
+        String length = parts.findAttributeValue("monthWidth", "type"); // wide, abbreviated, narrow
+        if (length.equals("narrow")) {
+            return null; // no examples for narrow
+        }
+        String context = parts.findAttributeValue("monthContext", "type"); // format, stand-alone
+        String calendarId =
+                parts.findAttributeValue("calendar", "type"); // gregorian, islamic, hebrew, ...
+        String monthNumId =
+                parts.findAttributeValue("month", "type"); // 1-based: 1, 2, 3, ... 12 or 13
+
+        final String[] skeletons = {"yMMMMd", "yMMMd", "yMMMM", "yMMM"};
+        int skeletonIndex = (length.equals("wide")) ? 0 : 1;
+        if (!context.equals("format")) {
+            skeletonIndex += 2;
+        }
+        String checkPath =
+                "//ldml/dates/calendars/calendar[@type=\""
+                        + calendarId
+                        + "\"]/dateTimeFormats/availableFormats/dateFormatItem[@id=\""
+                        + skeletons[skeletonIndex]
+                        + "\"]";
```

Comment:
I wonder if we should have an XPathParts Builder designed for constructing XPaths in a loop with syntax checking (i.e. that the xpath remains valid).

Reply:
How many places would we use something like that?